test-validator/src/lib.rs

```diff
@@ -782,25 +787,65 @@ pub struct TestValidator {
 }
 
 impl TestValidator {
-    /// Create and start a `TestValidator` with no transaction fees and minimal rent.
-    /// Faucet optional.
-    ///
-    /// This function panics on initialization failure.
-    pub fn with_no_fees(
+    /// Helper to block on async waiting logic
+    fn block_on_wait(&self, wait_for_fees: bool) {
+        let runtime = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(2)
+            .enable_io()
+            .enable_time()
+            .build()
+            .unwrap();
```

Comment:
nit: I didn't test, but does this need to be multi-threaded? I ask because the runtime is just doing a couple of waits and then getting destroyed, so maybe we can go with a lighter runtime